public/scripts/index.js

```diff
@@ -0,0 +1,171 @@
+        }
+        draw(currentFrame, context, cellSize);
+        spanMouseX.innerHTML = 'Column ' + point2.x;
+        spanMouseY.innerHTML = 'Row ' + point2.y;
+    }
+    else {
+        spanMouseX.innerHTML = 'Out of bounds';
+        spanMouseY.innerHTML = 'Out of bounds';
+    }
+}
+function getInputValue() {
+    rows = document.getElementById("rows").value;
+    cols = document.getElementById("columns").value;
+
+    const h = rows * cellSize;
+    canvas.height = h;
+
+    const w = cols * cellSize;
+    canvas.width = w;
+
+    const arrH = h / cellSize;
+    const arrW = w / cellSize;
+
+    currentFrame = makeMatrix(arrH, arrW);
+    nextFrame = makeMatrix(arrH, arrW);
+    intermediateFrame = makeMatrix(arrH, arrW);
+
+    let context = canvas.getContext("2d");
+
+    drawBoard(w, h, context, cellSize);
+}
+
+function simulate(){
+    let intermediateFrame = getIntermediateFrame(currentFrame, rows, cols);
+    for (let i = 0; i < intermediateFrame.length; i++){
+        for (let j = 0; j < intermediateFrame[0].length; j++){
+                nextFrame[i][j] = calcNewCellState(currentFrame[i][j], intermediateFrame[i][j]);
+        }
+    }
+    currentFrame = nextFrame;
+    draw(currentFrame, context, cellSize);        
+}
+
+function drawPreset(){
+    let value = getPreset();
+    let numValue = parseInt(value);
+    if (numValue === 1 && currentFrame.length >= 17 && currentFrame[16][16] != undefined ){
+        clear();
+        currentFrame[8][7] = 1;
+        currentFrame[8][8] = 1;
+        currentFrame[8][9] = 1;
+        currentFrame[9][8] = 1;
+        currentFrame[9][7] = 1;
+        currentFrame[9][6] = 1;
+        draw(currentFrame, context, cellSize);
+    }
+    if (numValue === 2 && currentFrame.length >= 17 && currentFrame[16][16] != undefined){
+        clear(currentFrame, intermediateFrame, nextFrame);
+        currentFrame[7][8] = 1;
+        currentFrame[8][9] = 1;
+        currentFrame[8][10] = 1;
+        currentFrame[8][11] = 1;
+        currentFrame[8][6] = 1;
+        currentFrame[8][5] = 1;
+        currentFrame[6][6] = 1;
+        draw(currentFrame, context, cellSize);
+    }
+    if (numValue === 3 && currentFrame.length >= 17 && currentFrame[16][16] != undefined){
+        clear(currentFrame, intermediateFrame, nextFrame);
+        currentFrame[7][9] = 1;
+        currentFrame[7][6] = 1;
+        currentFrame[8][10] = 1;
+        currentFrame[9][10] = 1;
+        currentFrame[9][6] = 1;
+        currentFrame[10][7] = 1;
+        currentFrame[10][8] = 1;
+        currentFrame[10][9] = 1;
+        currentFrame[10][10] = 1;
+        draw(currentFrame, context, cellSize);
+    }
+
+    if (currentFrame === undefined){
+        alert('you must draw a grid')
+    }
+    else if (numValue === 0){
+        alert('You must select a preset');
+    }
+    else if (currentFrame.length < 17){
+        alert('Grid is too small. The minimum size is 17x17');
+    }    
+}
+function clear(){
```

Comment:
If you think about it clear is just making a matrix filled with zeros an you alredy have a func for that use case